Servers/UI/OJS.Servers.Ui/ClientApp/src/components/guidelines/forms/FormControl.tsx

```diff
@@ -81,122 +89,168 @@ const FormControl = ({
     onChange = null,
     /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
     onInput = null,
+    onKeyDown,
+    onClick,
     className = '',
     containerClassName = '',
     labelText = '',
     labelClassName = '',
     type = FormControlType.text,
     checked = false,
     id = generateId(),
+    disableLabel = false,
 }: IFormControlProps) => {
     const [ formControlValue, setFormControlValue ] = useState(value);
+
     const [ isChecked, setIsChecked ] = useState<boolean>(checked);
 
     const componentClassName = concatClassNames(type !== FormControlType.checkbox
         ? styles.formControl
         : null, className);
 
-    const handleOnChange = (ev: ChangeEvent<TextAreaOrInputElement>) => {
-        if (type === FormControlType.checkbox) {
-            setIsChecked(!isChecked);
+    const handleOnChange = useCallback(
+        (ev: ChangeEvent<TextAreaOrInputElement>) => {
+            if (type === FormControlType.checkbox) {
+                setIsChecked(!isChecked);
 
-            return;
-        }
+                return;
+            }
 
-        setFormControlValue(ev.target.value);
+            setFormControlValue(ev.target.value);
 
-        if (isNil(onChange)) {
-            return;
-        }
+            if (isNil(onChange)) {
+                return;
+            }
 
-        onChange(ev.target.value as string);
-    };
+            onChange(ev.target.value as string);
+        },
+        [ isChecked, onChange, type ],
+    );
 
-    const handleOnInput = (ev: FormEvent<TextAreaOrInputElement>) => {
-        const element = ev.target as TextAreaOrInputElement;
+    const handleOnInput = useCallback(
+        (ev: FormEvent<TextAreaOrInputElement>) => {
+            const element = ev.target as TextAreaOrInputElement;
 
-        setFormControlValue(element.value);
+            setFormControlValue(element.value);
 
-        if (isNil(onInput)) {
-            return;
-        }
+            if (isNil(onInput)) {
+                return;
+            }
 
-        onInput(element.value);
-    };
+            onInput(element.value);
+        },
+        [ onInput ],
+    );
```

Comment:
Don't change the indentation of already implemented functions into random PR for SearchPage
It's not a requirement of the issue.

Reply:
ESLint automatically updated them, because I added useCallback.